lms/djangoapps/course_blocks/transformers/utils.py

```diff
@@ -18,6 +19,67 @@ def get_field_on_block(block, field_name, default_value=None):
     return default_value
 
 
+def collect_nearest_subsection(
+        block_structure,
+        transformer,
+):
+    """
+    Does magic. 2 passes. Maybe I'll generalize this, idk.
```

Comment:
Warning: Everything in this method is just-barely-at-the-proof-of-concept complete, there's a lot of cleanup to be done before it's ready for review.